### ISSUE TITLE
Accuracy issues inferencing bert-base-uncased-onnx-16-int8

### ISSUE THREAD

**Issue by cjvolzka** [onnx/onnx-mlir#2566]
Using the attached script (get_hg_model.py), I tried running the [bert-base-uncased](https://huggingface.co/bert-base-uncased) model from hugging face through onnx-mlir. I also tried a version of the model quantized (int8) by onnx runtime.

Afterward the model compiles and runs fine in onnx-mlir. I compared the onnx-mlir output with output from the same model using onnx runtime (bert_ort_output.py) and the values matched within an absolute tolerance of  0.000044.

However when I tried the int8 version of the model, onnx-mlir had a notable accuracy loss compared to onnx-runtime. The onnx-mlir output was off by an a_tol of 0.363081 which is way more than is expected when compiling for CPU.

I used the following as input to the model (non-quantized and quantized versions)
```
inputs = {
    'input_ids': np.array([[ 101, 1996, 3125, 1997, 2166, 2003,  103, 1012,  102]], dtype=np.int64),
    'attention_mask': np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=np.int64),
    'token_type_ids': np.array([[1, 1, 1, 1, 1, 1, 1, 1, 1]], dtype=np.int64),
}
```

**Comment by AlexandreEichenberger** [comment]
@cjvolzka wonder if you don't have to copy/paste the code. I don't see the attachment

**Comment by cjvolzka** [comment]
I did some investigation into this. There's not really a bug that I can see.

What's happening is that nodes before DynamicQuantizeLinear (DQL) have minor output differences compared to ORT. Nothing major and well under the tight CPU tolerances we target. However those minor differences have a noticeable affect once we reach DQL. 

If the smallest or largest value of the input is affected, the calculated scale for quantization changes. A subtle change in the scale causes some values to round up or down differently when quantized to ints.

In this model in particular, some values round down to the zero offset value (or some others that round to the zero offset value in ORT, now run up or down to be non-zero when quantized). Moving to or away from zero followed by the MatMul Integer is what is causing the notable tolerance difference. 

If the DQL is followed by a Softmax or some other form of normalization, the CPU tolerances fall back within expectation. Other DQL's in the model are like this. The trick is that for the final DQL, there isn't a normalization so I'm seeing larger than normal toleration differences on CPU. 

So they are large for a CPU model, but nothing appears to be broken. 

If I take the exact values generated by onnx-mlir to a DQL run in ORT, ORT gives the same results onnx-mlir does for DQL. In the reverse, if I give the ORT intputs to onnx-mlir DQL, I also get the same output. So there's nothing wrong or different in our DQL implementation. It's just the nature of the accumulation of differences rounding and math implementations used.

**Comment by soldior777** [comment]
I wanted to thank you for looking into this matter very grateful.

SV55 216 Schok
